### PR TITLE
Add CloudFormation template files

### LINKED ISSUE #16: Include AWS CodeStar/CloudFormation support
It would be great if the boilerplate included template.yml and buildspec.yml to integrate with downstream dependencies like CodePipeline/CodeDeploy.

### PR DESCRIPTION
**Pull request by irvinlim**
Fixes #16.

### CODE REVIEW

File: buildspec.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: 0.1
+phases:
+  install:
+    commands:
+      - npm install
+      - npm run package
+      - aws cloudformation package --template-file samTemplate.yml 
+                                   --s3-bucket codepipeline-es2017-lambda-boilerplate
```
Comment (ChadMoran) [inline buildspec.yml]: CodeBuild supports environment variables, perhaps we could follow the convention CodeStar uses when it creates the CodeBuild step and use $S3_BUCKET which can be configured in the CodeBuild UI.

**Comment by ChadMoran** [comment]
:shipit:

**Comment by irvinlim** [comment]
Alright, thanks! 

The behaviour is slightly bugged out when using the CodePipeline wizard to create the CodeBuild project though - I had to manually set the artifact upload location through CodeBuild to specify the S3 bucket name, even after following the [CodePipeline tutorial](http://docs.aws.amazon.com/lambda/latest/dg/build-pipeline.html). Perhaps a point to note for other users who might be using CodePipeline!

**Comment by ChadMoran** [comment]
That's a good point, I'll see if there's a way around that and submit a PR.  Thanks for doing this.

**Comment by irvinlim** [comment]
No problem! Thanks for the hard work at AWS! (I assume you're working there)

**Pull request merged**